agent/taskresource/cgroup/cgroup.go

```diff
@@ -97,12 +109,95 @@ func (c *CgroupResource) GetName() string {
 	return resourceName
 }
 
+// DesiredTerminal returns true if the cgroup's desired status is REMOVED
+func (c *CgroupResource) DesiredTerminal() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.GetDesiredStatus() == taskresource.ResourceStatus(CgroupRemoved)
+}
+
+// KnownCreated returns true if the cgroup's known status is CREATED
+func (c *CgroupResource) KnownCreated() bool {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+
+	return c.GetKnownStatus() == taskresource.ResourceStatus(CgroupCreated)
+}
+
+// TerminalStatus returns the last transition state of cgroup
+func (c *CgroupResource) TerminalStatus() taskresource.ResourceStatus {
```

Comment:
This needs a lock

Reply:
This method just returns `CgroupRemoved`. Hence not using any lock.